pennylane/operation.py

```diff
@@ -262,6 +262,62 @@ def _matrix(*params):
         """
         raise NotImplementedError
 
+    @property
+    def matrix(self):
+        r"""Matrix representation of an instantiated operator
+        in the computational basis.
+
+        **Example:**
+
+        >>> U = qml.RY(0.5, wires=1)
+        >>> U.matrix
+        >>> array([[ 0.96891242+0.j, -0.24740396+0.j],
+                   [ 0.24740396+0.j,  0.96891242+0.j]])
+
+        Returns:
+            array: matrix representation
+        """
+        return self._matrix(*self.parameters)
+
+    @classmethod
+    def _eigvals(cls, *params):
+        """Eigenvalues of the operator in the computational basis.
```

Comment:
What does it mean to return the eigenvalues of an operator in the computational basis? Every operator has it's own eigenbasis :thinking:

Reply:
I'm also not very happy with this formulation. But both `[-1, 1]` and `[1, -1]` would be viable eigenvalues of `Z`, but I need the latter for the diagonal gates. Maybe "matching the order of computational basis vectors when diagonalized"? Or something like this?